pydantic/config.py

```diff
@@ -709,11 +709,30 @@ class Model(BaseModel):
 
     defer_build: bool
     """
-    Whether to defer model validator and serializer construction until the first model validation.
+    Whether to defer model validator and serializer construction until the first model validation. Defaults to False.
 
     This can be useful to avoid the overhead of building models which are only
     used nested within other models, or when you want to manually define type namespace via
-    [`Model.model_rebuild(_types_namespace=...)`][pydantic.BaseModel.model_rebuild]. Defaults to False.
+    [`Model.model_rebuild(_types_namespace=...)`][pydantic.BaseModel.model_rebuild].
+
+    See also [`defer_build_mode`][pydantic.config.ConfigDict.defer_build_mode].
+
+    !!! note
+        `defer_build` does not work by default with FastAPI Pydantic models. Meaning the validator and serializer
+        is constructed immediately when the model is used in FastAPI routes instead of construction being deferred.
+        You also need to use [`defer_build_mode='always'`][pydantic.config.ConfigDict.defer_build_mode] with
+        FastAPI models.
+    """
+
+    defer_build_mode: Literal['only_model', 'always']
```

Comment:
Hmm, I wonder if there's a more intuitive second option that we could use that explicitly mentions type adapters...

Reply:
Another option would be to make it a `tuple[Literal['model', 'type_adapter'], ...]` instead of it being `Literal['model', 'model_and_type_adapter']` 🤔

Reply:
Or `set[Literal['model', 'type_adapter']]`

WDYT?

Reply:
I now went with `tuple[Literal['model', 'type_adapter'], ...]`

Reply:
That feels more intuitive to me, thanks!

Reply:
The caveat with this is that user could remove the `model` from tuple 😅 So either we somewhere validate that `model` must be present or support skipping defer building of model in that case 😅 Another option is to instead have `_defer_build_additional: tuple['type_adapter'] | None`